### PR TITLE
Add libreoffice-24.8

### PR DESCRIPTION
**Pull request by smoser**
Things to note
 * bash-binsh - configure has bashisms but is marked as /bin/sh
 * drop runtime depends, melange picks them up.
 * libnss is needed, dlopened at runtime.
 * see https://github.com/wolfi-dev/os/pull/31210 for more
   information on system-boost

### CODE REVIEW

File: libreoffice-24.8.yaml [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  - uses: git-checkout
+    with:
+      repository: https://github.com/LibreOffice/core
+      tag: libreoffice-${{package.version}}
+      expected-commit: 0f794b6e29741098670a3b95d60478a65d05ef13
+
+  # patch rather than cherry-pick. The git fetch of main takes multiple minutes.
+  - uses: patch
+    with:
+      patches: 0001-raptor-Use-without-www.patch
```
Comment (xnox) [inline libreoffice-24.8.yaml:106]: @smoser this patch is missing i think. symlink or copy the dir from previous libreoffice?

Comment (smoser) [reply libreoffice-24.8.yaml:106]: Doh. Thanks. I'll fix

Comment (smoser) [reply libreoffice-24.8.yaml:106]: done.

**Comment by smoser** [comment]
I tried to use the system boost here. if the test does not pass, then we'll switch back to the builtin.

**Comment by smoser** [comment]
https://github.com/wolfi-dev/os/actions/runs/11412511156/job/31758616273#step:8:1
failed when running `--help` with:
```
2024/10/19 00:45:55 INFO running the main test pipeline
2024/10/19 00:45:55 WARN + '[' -d /home/build ]
2024/10/19 00:45:55 WARN + cd /home/build
2024/10/19 00:45:55 WARN + /usr/lib/libreoffice/program/soffice.bin --help
2024/10/19 00:45:55 WARN /usr/lib/libreoffice/program/soffice.bin: error while loading shared libraries: libnss3.so: cannot open shared object file: No such file or directory
2024/10/19 00:45:55 INFO pod e33a1e4dd7895a274fea392b191e50c5aec4769379cffa595c46a20e2b1cd5dd terminated
2024/10/19 00:45:55 INFO ERROR: failed to test package. the test environment has been preserved:
2024/10/19 00:45:55 INFO   workspace dir: /temp/melange-workspace-2047042158
2024/10/19 00:45:55 INFO   guest dir: /temp/melange-guest-3036578255
2024/10/19 00:45:55 ERRO failed to test package: unable to run pipeline: unable to run pipeline: task exited with code 127
make: *** [Makefile:136: test/libreoffice-24.8] Error 1
```

**Comment by smoser** [comment]
Bummer. Looks like we either need to drop system boost, or find a fix. 

    2024/10/19 02:10:09 WARN Unspecified Application Error name="convert a csv to pdf"

**Review by xnox — APPROVED** [review]

**Pull request merged**